src/components/Footer/Footer.tsx

```diff
@@ -0,0 +1,68 @@
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+import { useEffect, useState } from 'react';
+import { FilterType } from '../../types/FilterType';
+
+interface FooterProps {
+  todos: Todo[];
+  todosType: FilterType;
+  handleTodosTypeChange: (todosType: FilterType) => void;
+  handleDeleteTodo: (todoId: number) => Promise<void>;
+}
+
+export const Footer: React.FC<FooterProps> = ({
+  todos,
+  todosType = FilterType.ALL,
+  handleTodosTypeChange,
+  handleDeleteTodo,
+}) => {
+  const [active, setActive] = useState(0);
+  const [completed, setCompleted] = useState<Todo[]>([]);
+
+  useEffect(() => {
+    setActive(todos.filter(todo => !todo.completed).length);
+    setCompleted(todos.filter(todo => todo.completed));
+  }, [todos]);
```

Comment:
```suggestion
  const active = todos.filter(todo => !todo.completed).length;
  const completed = todos.filter(todo => todo.completed);
```
it just can be the vars